mobile/lib/features/wallet/share_invoice_screen.dart

```diff
@@ -84,8 +84,16 @@ class _ShareInvoiceScreenState extends State<ShareInvoiceScreen> {
                         });
 
                         final router = GoRouter.of(context);
+                        const localCoordinatorPubkey = "02dd6abec97f9a748bf76ad502b004ce05d1b2d1f43a9e76bd7d85e767ffb022c9";
                         try {
-                          await openCoordinatorChannel(config.host);
+                          if ( config.coordinatorPubkey == localCoordinatorPubkey) {
+                            await payInvoiceWithFaucet(widget.invoice);
+                          }
+                          else {
+                            // For remote coordinator, we open the channel
```

Comment:
🔧 I wouldn't go for a different setup on our regtest setup than on our localhost setup. IMHO that's only confusing as we would have 3 different ways on how the wallet gets funded. Ideally we should always go the pay invoice path to have it as close as possible to production.

If we feel we can go that path for our local setup, why not also change it for our regtest setup?

Reply:
This way brings it closer to production in local testing, so it's a major step forward.

I clearly explained that this is needed for 
- specifying custom amount to be sent,
- seeing it in transaction history.

(I need the first one for testing a bug I'm working on, so I'm proceeding).

As for not changing it for regtest - I wanted to be conservative.  Earlier I opted to do the same for the regtest, but it got changed due to us closing the channels with LND randomly on and the team voted for having something that worked more closely. I have no capacity to debug it, nor we're actively pushing 10101-test right now, so this is low on my priority list.
I would be happy if this happened someday though, because I feel it was a hack that we went with bypassing paying the invoice at all.

TL;DR: I'm removing this terrible workaround from the most used path, so we can test more. I hope that we can remove it everywhere.